add information on customer managed resoures (#4268)
aws-amplify/docs: src/pages/cli/custom/cdk.mdx

> Note: Always append the Amplify environment name to a resource name to ensure correct behavior with Amplify's multi-environment workflows.

<Callout warning>

> When creating custom resources using Amplify CLI, the CLI will require additional permissions outside the Amplify managed policy.

Reviewer: ```suggestion
When creating custom resources using Amplify CLI, the CLI may require additional permissions outside the Amplify managed policy, `AdministratorAccess-Amplify`.
```

> When creating custom resources using Amplify CLI, the CLI may require additional permissions outside the Amplify managed policy, `AdministratorAccess-Amplify`.

Reviewer: ```suggestion
When creating custom resources using Amplify CLI, the CLI may require additional permissions outside the Amplify managed policy, [`AdministratorAccess-Amplify`](https://console.aws.amazon.com/iam/home#policies/arn:aws:iam::aws:policy/AdministratorAccess-Amplify).
```

> Please refer to the Amplify [IAM Policy](/cli/reference/iam) documentation for additional information on Amplify managed Policy.

Reviewer: ```suggestion
Please refer to the Amplify [IAM Policy](/cli/reference/iam) documentation for additional information regarding the necessary permissions for Amplify CLI.
```
